Kernel/Graphics/FramebufferDevice.cpp

```diff
@@ -40,6 +40,15 @@ ErrorOr<Memory::Region*> FramebufferDevice::mmap(Process& process, OpenFileDescr
     if (range.size() != framebuffer_length)
         return EOVERFLOW;
 
+    ArmedScopeGuard scope_guard = [&] {
+        if (m_real_framebuffer_region)
+            m_real_framebuffer_region = nullptr;
+        if (m_swapped_framebuffer_region)
+            m_swapped_framebuffer_region = nullptr;
+        // Note: Creation of m_userspace_framebuffer_region does not have to be
+        //       corrected due to it being the last fallible operation
+    };
```

Comment:
Perhaps a more linear approach with temporaries would make this a bit more readable:
```cpp
auto userspace_real_framebuffer_vmobject = TRY(Memory::AnonymousVMObject::try_create_for_physical_range(m_framebuffer_address, framebuffer_length));
auto real_framebuffer_vmobject = TRY(Memory::AnonymousVMObject::try_create_for_physical_range(m_framebuffer_address, framebuffer_length));
auto swapped_framebuffer_vmobject = TRY(Memory::AnonymousVMObject::try_create_with_size(framebuffer_length, AllocationStrategy::AllocateNow));
auto real_framebuffer_region = TRY(MM.allocate_kernel_region_with_vmobject(*m_real_framebuffer_vmobject, framebuffer_length, "Framebuffer", Memory::Region::Access::ReadWrite));
auto swapped_framebuffer_region = TRY(MM.allocate_kernel_region_with_vmobject(*m_swapped_framebuffer_vmobject, framebuffer_length, "Framebuffer Swap (Blank)", Memory::Region::Access::ReadWrite));

m_userspace_real_framebuffer_vmobject = move(userspace_real_framebuffer_vmobject);
m_real_framebuffer_vmobject = move(real_framebuffer_vmobject);
m_swapped_framebuffer_vmobject = move(swapped_framebuffer_vmobject);
m_real_framebuffer_region = move(real_framebuffer_region);
m_swapped_framebuffer_region = move(swapped_framebuffer_region);
```

Etc.. By setting everything up in temporaries, we avoid the need for a scope guard in the first place.

Reply:
This is a lot nicer!
In addition to @IdanHo's suggestion, I have moved this to `try_to_initialize` to remove some duplication